src/model/paths.ml

```diff
@@ -255,23 +259,27 @@ module Identifier = struct
     type t_pv = Id.container_page_pv
   end
 
+  module NonSrc = struct
+    type t = Paths_types.Identifier.non_src
+    type t_pv = Paths_types.Identifier.non_src_pv
+  end
+
   module SourceDir = struct
     type t = Id.source_dir
     type t_pv = Id.source_dir_pv
     let equal = equal
     let hash = hash
     let compare = compare
-    let rec name = function
-      | { iv = `SourceDir (p, n); _ } -> name p ^ n ^ "/"
-      | { iv = `SourceRoot _; _ } -> "./"
   end
 
   module SourcePage = struct
     type t = Id.source_page
     type t_pv = Id.source_page_pv
-    let equal = equal
-    let name { iv = `SourcePage (p, name); _ } = SourceDir.name p ^ name
-    
+  end
+
+  module SourceLocation = struct
```

Comment:
What do you think of removing the `Paths` module ?

It slowed us down significantly while implementing source code rendering with @panglesd.
It's only made of aliases and is annoying to edit (a lot of boilerplate).
It makes `Paths_types` harder to understand and makes Merlin's printed types harder to read.

With all these downsides, I'd argue that it's not worth the slightly improved paths: `Id.SourceLocation.t` instead of `Id.source_location`.
Maps are almost only used in `Component`, which could as well define them inline.

Reply:
Yeah, I'm leaning in that direction too, especially now that most of the additional functions added in the namespace modules hve gone.